File: 1. Fast and slow pointers/question 234.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        slow=fast=head
        while fast and fast.next:
            fast=fast.next.next
            slow=slow.next
        
        node=None
        while slow:
            temp = slow.next
            slow.next = node
            node = slow
            slow = temp
        
        front,back = head,node
        while back:
            if front.val!=back.val:
                return False
            front=front.next
            back=back.next
        return True

File: 1. Fast and slow pointers/question 234.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
    let slow = head, fast = head

    while (fast !== null && fast.next !== null){
        fast = fast.next.next
        slow = slow.next
    }
    let newNode = null
    while (slow!==null){
        let temp = slow.next
        slow.next = newNode
        newNode = slow
        slow = temp
    }
    let start = head
    while (newNode!==null){
        if (newNode.val!==start.val) {return false}
        start = start.next
        newNode = newNode.next
    }
    return true
};